JapaneseAddress/Block/Checkout/BillingAddressComponent.php

```diff
@@ -16,11 +18,36 @@
  */
 class BillingAddressComponent implements LayoutProcessorInterface
 {
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     *
+     */
+    const XML_PATH_STORE_LOCALE = 'general/locale/code';
+
+    /**
+     * BillingAddressComponent constructor.
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig
+    )
+    {
+        $this->scopeConfig = $scopeConfig;
+    }
+
     /**
      * @inheritDoc
      */
     public function process($jsLayout)
     {
+        $locale = $this->getStoreLocale();
+        if($locale != 'ja_JP') {
```

Comment:
Let's move this check and `getStoreLocale` method to the separate class with method like `checkJapaneseLocale`. It looks like it might be widely reused.

Reply:
Hmm, but no other code use similar condition 🤔